File: v2/DSFile.cpp
/*
 * DSFile.cpp
 *
 *  Created on: 24 Jun 2011
 *      Author: James Osler
 */

#include "DSFile.h"

DSFile::DSFile(string name, string directory)
{
	setName(name);
	setDirectory(directory);
	version = 1;

}

DSFile::~DSFile()
{
}

string DSFile::getDirectory() const
{
    return directory;
}

string DSFile::getExtension() const
{
    return extension;
}

string DSFile::getName() const
{
    return name;
}

string DSFile::getFileString() const
{
	if(getExtension() != "")
		return getName() + "." + getExtension();
	else
		return getName();
}

int DSFile::getVersion() const
{
    return version;
}

void DSFile::setDirectory(string directory)
{
    this->directory = directory;
}

void DSFile::setName(string name)
{
	if(name.find(".") == string::npos) // no '.' in name
	{
		this->name = name;
		this->extension = "";
	}
	else
	{
		size_t last = name.rfind("."); // POW!!!
		this->extension = name.substr(last+1);
		this->name = name.substr(0,name.size()-(this->extension.size() + 1));
	}
}

void DSFile::setVersion(int version)
{
    this->version = version;
}

File: v2/dsfile.py
#!/usr/bin/env python

class dsfile():
	"""
	dropsync file
	"""
	
	def __init__(self,name,directory):
		self.setName(name)
		self.directory = directory
		self.version = 1
		
	def getFileString(self):
		if self.extension != "":
			return self.name + "." + self.extension
		else:
			return self.name
		
	def setName(self, name):
		last = name.find(".")
		if(last == -1): # not present
			self.name = name
			self.extension = ""
		else:
			last = name.rfind(".")
			if(last != 0): # normal and hidden with ext
				self.extension = name[last+1:]
				self.name = name[:len(name)-(len(self.extension) + 1)]
			else: # hidden files with no ext
				self.extension = ""
				self.name = name

if __name__ == "__main__":
	da = dsfile("moo1","/ome/jamie/")
	db = dsfile("moo2.ext","/home/jamie/")
	dc = dsfile("moo2.midl.ext","/home/jamie/")
	dd = dsfile(".moo2","/home/jamie/")
	de = dsfile(".moo2.ext","/home/jamie/")

	print da.getFileString()
	print db.getFileString()
	print dc.getFileString()
	print dd.getFileString()
	print de.getFileString()


	
